README.md

```diff
@@ -1,3 +1,197 @@
+
+We can use it this way:
+
+```javascript
+import { ConsentString } from "consent-string";
+
+const consents = new ConsentString();
+
+/**
+ * we have to register this and be sure that our vendor list is at a correct format
+ */
+consents.setGlobalVendorList(ourGlobalVendorList);
+
+/**
+ * Giving current user consent to a specific vendor or purpose
+ */
+consents.setPurposeAllowed(1, true);
+consents.setVendorAllowed(287, true);
+```
+
+The `ConsentString` aims to be shared with the vendors that try to get user information. When these vendors will "ask" (see CMP part) for consents, this `ConsentString` will be shared with them. If they are concerned, they can collect information. Else, they can't.
+
+### Vendor based / Purpose based / Feature based
+
+It's possible to manage granularity for consent management. Some website prefers to let the user chose which `vendor` are authorized to gather information.
+Some other, like us at M6Web, prefers to use a less granular one and to manage consents at a purpose level.
+
+**Example of a vendor based consent management:**
+
+- [x] Google Analytics
+- [ ] Krux
+
+**Example of a vendor based consent management:**
+
+- [ ] Information storage and access
+- [x] Personalisation
+
+_Feature based is not available for now with IAB Consent Framework. But since the feature field is available in the VendorList, we can imagine managing consents using feature based approach_
+
+## Consent Management Provider (CMP)
+
+"CMP" means a company that can read the vendors chosen by a website operator and the consent status of an end user (either service specific (through a first-party cookie) or global (through a third-party cookie).
+
+### Required API
+
+Every consent manager MUST provide the following API:
+
+`__cmp(Command, Parameter, Callback)`
+
+This API provide few command : 
+
+<table>
+  <tr>
+    <td>Command</td>
+    <td>Parameter</td>
+    <td>Callback</td>
+    <td>Comments</td>
+  </tr>
+  <tr>
+    <td>getVendorConsents</td>
+    <td>Vendors Ids [Array:String]</td>
+    <td>Callback(VendorConsents object, success: boolean)</td>
+    <td>Vendor consents strings</td>
+  </tr>
+  <tr>
+    <td>getConsentData</td>
+    <td>Consent string [String]</td>
+    <td>Callback(VendorConsentData object, success: boolean)</td>
+    <td>Consents data from user</td>
+  </tr>
+  <tr>
+    <td>ping</td>
+    <td> - </td>
+    <td>Callback(PingReturn object, success: boolean)</td>
+    <td>The "ping" command invokes the callback immediately with information about whether the main CMP script has loaded yet and if GDPR has been configured for all users or just EU users. </td>
+  </tr>
+</table>
+
+### Communication with vendors
+
+For communication with vendors we mostly use the `__cmp` function for local communication and via `iframe` (standard).
+
+#### `window.__cmp`
+The function __cmp is created like this : 
+```js
+export const __cmp = {
+  getVendorConsents: vendorConsentsLoader,
```

Comment:
WYT of renaming `vendorConsentsLoader` in something that people can understand? Since it's a public project, everybody should be able to understand this piece of code.

Something more open / generic like:

```javascript
getVendorConsents: () => /* some stuff concerning the vendors consents */
```

would help people understand